src/drt/src/db/tech/frConstraint.h

```diff
@@ -315,13 +315,27 @@ class frMinEnclosedAreaConstraint : public frConstraint
 {
  public:
   // constructor
-  frMinEnclosedAreaConstraint(frCoord areaIn) : area(areaIn), width(-1) {}
+  frMinEnclosedAreaConstraint() : rule_(nullptr) {}
   // getter
-  frCoord getArea() const { return area; }
-  bool hasWidth() const { return (width != -1); }
-  frCoord getWidth() const { return width; }
+  odb::dbTechMinEncRule* getDbTechMinEncRule() const { return rule_; }
+  frCoord getArea() const
+  {
+    frUInt4 minEnclosedArea;
+    rule_->getEnclosure(minEnclosedArea);
+    return minEnclosedArea;
+  }
+  bool hasWidth() const
+  {
+    frUInt4 width;
+    return rule_->getEnclosureWidth(width);
+  }
+  frCoord getWidth() const
+  {
+    frUInt4 width;
+    return (rule_->getEnclosureWidth(width)) ? width : -1;
+  }
   // setter
-  void setWidth(frCoord widthIn) { width = widthIn; }
+  void setDbTechMinEncRule(odb::dbTechMinEncRule* ruleIn) { rule_ = ruleIn; }
```

Comment:
This should be a ctor argument and this method removed.  Same in the other constraints.